src/error.rs

```diff
@@ -135,7 +135,7 @@ impl Error {
 
     #[allow(unused)]
     pub(crate) fn into_io(self) -> io::Error {
-        io::Error::new(io::ErrorKind::Other, self)
+        self.into()
     }
```

Comment:
Since this is only crate-public, then we should just delete this method competely and replace it with .into()

Reply:
agreed but @seanmonstar seemed a bit dubious on this change so figured I'd just start with the bare minimum and see where the discussion went 😄